### PR TITLE
Bump AWSLC_API_VERSION for EVP_chacha20_poly1305

### PR DESCRIPTION
**Pull request by andrewhop**
### Description of changes: 
Bump API version for our new EVP_CIPHER version of ChaCha. Also add a test of speed.cc with the 2022 branch of AWS-LC. 

### Testing:
This adds a new build to make sure it continues to work. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tests/ci/run_benchmark_build_tests.sh [hunk 1]
```diff
@@ -4,61 +4,78 @@
...
+scratch_folder=${SYS_ROOT}/"benchmark-scratch"
+install_dir="${scratch_folder}/libcrypto_install_dir"
 openssl_url='https://github.com/openssl/openssl.git'
 openssl_1_1_branch='OpenSSL_1_1_1-stable'
 openssl_1_0_branch='OpenSSL_1_0_2-stable'
-# OpenSSL 3.0 branch still isn't stable yet, so we lock down
-# a tag to prevent build failures that they introduced from
-# failing our CI.
-openssl_3_0_branch='openssl-3.0.5'
+openssl_3_0_branch='openssl-3.1'
```
Comment (torben-hansen) [inline tests/ci/run_benchmark_build_tests.sh:12]: Variable named `3.0` but really `3.1`.

Does this reflect when capturing and graphing later on in the workflow?

Comment (andrewhop) [reply tests/ci/run_benchmark_build_tests.sh:12]: Yeah, it uses the name of the branch in the metric. I can fix this and add a 3.2 test.

**Review by torben-hansen — DISMISSED** [review]

File: tool/speed.cc [hunk 2]
```diff
@@ -33,9 +33,23 @@
 
 #include "internal.h"
 
-#if !defined(OPENSSL_BENCHMARK)
+#include <openssl/crypto.h>
+#if defined(OPENSSL_IS_AWSLC)
+#include "bssl_bm.h"
+#elif defined(OPENSSL_IS_BORINGSSL)
+#define ORINGSSL_BENCHMARK
```
Comment (nebeid) [inline tool/speed.cc:40]:
```suggestion
#define BORINGSSL_BENCHMARK
```

File: tests/ci/run_benchmark_build_tests.sh [hunk 3]
```diff
@@ -4,90 +4,100 @@
...
+#    echo "building OpenSSL 3.0"
+#    git clone --depth 1 --branch "${openssl_3_0_branch}" "${openssl_url}" "${scratch_folder}/openssl-3.0"
+#    pushd "${scratch_folder}/openssl-3.0"
+#    mkdir -p "${install_dir}/openssl-3.0"
+#    ./Configure --prefix="${install_dir}/openssl-3.0" --openssldir="${install_dir}/openssl-3.0" -d
+#    make "-j${NUM_CPU_THREADS}" > /dev/null
+#    make install_sw
+#    popd
+#    rm -rf "${scratch_folder}/openssl-3.0"
+#}
```
Comment (WillChilds-Klein) [inline tests/ci/run_benchmark_build_tests.sh:70]: if this is made redundant by `build_openssl`, shall we just delete it?

File: tool/speed.cc [hunk 4]
```diff
@@ -2628,6 +2642,9 @@ bool Speed(const std::vector<std::string> &args) {
        !SpeedECMUL(selected) ||
        // OpenSSL 1.0 doesn't support Scrypt
        !SpeedScrypt(selected) ||
+#endif
+#if (!defined(OPENSSL_1_0_BENCHMARK) && !defined(BORINGSSL_BENCHMARK) && !defined(OPENSSL_IS_AWSLC)) || AWSLC_API_VERSION >= 24
+        // BoringSSL doesn't support ChaCha, OpenSSL 1.0 doesn't support ChaCha, only AWS-LC after API verison 24
```
Comment (WillChilds-Klein) [inline tool/speed.cc:2647]: minor nit: looks like BSSL [supports chacha](https://github.com/google/boringssl/blob/7460d7456433fbaf28ec679fc138fefd6b2120cd/include/openssl/aead.h#L126) through the EVP AEAD API, just not through the EVP Cipher API.

Comment (andrewhop) [reply tool/speed.cc:2647]: Yeah, this benchmark is specifically for EVP_chacha20_poly1305, AWS-LC/BoringSSL wil both test EVP_aead_chacha20_poly1305 down below. I'll update the comment to make it more clear.

**Review by WillChilds-Klein — APPROVED** [review]

**Review by skmcgrail — APPROVED** [review]

**Pull request merged**